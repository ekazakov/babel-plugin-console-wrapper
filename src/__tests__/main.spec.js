const path = require('path');
const babel = require('babel-core');
const plugin = require('../');

const transform = (fixtureName, options = {}) => {
    const fixture = `./src/__tests__/fixtures/${fixtureName}`;
    const { code } = babel.transformFileSync(fixture, {
        plugins: [[plugin, options]]
    });

    return code;
};

describe('Babel plugin console wrapper', () => {
    it('should transform with options', () => {
        const code = transform('simple.js', {
            fnName: 'testWrapper',
            fnImportPath: 'src/foo/bar/testWrapper',
            srcRoot: 'src'
        });
        expect(code).toMatchSnapshot();
    });

    it.todo('shouldn`t change code if found nothing');

    it.todo('should throw error if `fnName` option is not specified');
    it.todo('should throw error if `fnImportPath` option is not specified');
});
