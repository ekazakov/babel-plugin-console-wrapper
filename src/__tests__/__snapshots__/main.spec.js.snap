// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babel plugin console wrapper should transform with options 1`] = `
"import _testWrapper from '../../foo/bar/testWrapper';
function abc() {
    for (var i = 0; i < 100; i++) {
        _testWrapper(i, {
            line: 3,
            column: 8,
            filePath: './src/__tests__/fixtures/simple.js'
        });
    }

    return 42;
}

function xyz() {
    for (var i = 0; i < 100; i++) {
        _testWrapper(i, {
            line: 11,
            column: 8,
            filePath: './src/__tests__/fixtures/simple.js'
        });
    }

    if (true) {
        _testWrapper(['100', { foo: { y: [{ x: 1 }] } }], {
            line: 15,
            column: 8,
            filePath: './src/__tests__/fixtures/simple.js'
        });
    }

    otherFunction();
}

function outer() {
    function inner() {
        _testWrapper('does something', {
            line: 23,
            column: 8,
            filePath: './src/__tests__/fixtures/simple.js'
        });
    }

    return inner;
}"
`;
